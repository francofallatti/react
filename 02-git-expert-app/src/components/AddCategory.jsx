import PropTypes from 'prop-types'
import { useState } from 'react'

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('')

  const onInputChange = ({ target }) => {
    setInputValue(target.value)
  }

  const onSubmit = (event) => {
    event.preventDefault()
    const trimValue = inputValue.trim()
    if (trimValue.length <= 1) return

    onNewCategory(trimValue)
    setInputValue('')
  }

  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Buscar Gifs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
}
