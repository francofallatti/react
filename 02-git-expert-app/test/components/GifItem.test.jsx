import { render, screen } from '@testing-library/react'
import { GifItem } from '../../src/components/GifItem'

describe('Tests on <GifItem />', () => {
  const title = 'java'
  const url = 'https://www.image.com/java.gif'

  test('Snapshot test', () => {
    const { container } = render(<GifItem title={title} url={url} />)
    expect(container).toMatchSnapshot()
  })

  test('Test image URL', () => {
    render(<GifItem title={title} url={url} />)
    const { src, alt } = screen.getByRole('img')
    expect(src).toBe(url)
    expect(alt).toBe(title)
  })

  test('Test title in component', () => {
    render(<GifItem title={title} url={url} />)
    expect(screen.getByText(title)).toBeTruthy()
  })
})
