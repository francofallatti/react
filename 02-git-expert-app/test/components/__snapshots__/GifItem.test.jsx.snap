// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifItem /> Snapshot test 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="java"
      src="https://www.image.com/java.gif"
    />
    <p>
      java
    </p>
  </div>
</div>
`;
